guacamole/src/main/webapp/app/prompt/services/guacPrompt.js

```diff
@@ -0,0 +1,149 @@
+     */
+    service.getPrompt = function getPrompt() {
+        return storedPrompt();
+    };
+
+    /**
+     * Shows or hides the given prompt as a modal dialog.  Only one
+     * prompt dialog will be shown at a time.
+     *
+     * @param {Prompt|Boolean|Object} status
+     *     The prompt object to show.
+     */
+    service.showPrompt = function showPrompt(status) {
+        if (!storedPrompt() || !status)
+            storedPrompt(status);
+    };
+
+    /**
+     * Taking a list of prompts and a connection, display the prompt
+     * dialog for the user to fill out, and return a promise that
+     * responses will be provided.
+     *
+     * @param prompts
+     *     The list of prompts to display to the user.
+     *
+     * @param connection
+     *     The connection that the prompts are being used for.
+     *
+     * @returns
+     *     A promise for responses that the user will fill in.
+     */
+    service.getUserInput = function getUserInput(prompts,connection) {
+
+        var deferred = $q.defer();
+        var responses = {};
+        var homeUrl = '/';
+
+        if (prompts.length < 1)
+            deferred.resolve();
+
+        else {
+            service.showPrompt({
+                'title'     : 'Connection Parameters for ' + connection.name,
+                'connection' : connection,
+                'text'      : {
+                    key     : 'Please provide the following parameters to complete the connection:'
+                },
+                'prompts'   : prompts,
+                'actions'   : [{
+                    'name'  : 'Connect',
+                    'callback' : function() {
+                        deferred.resolve(responses);
+                        service.showPrompt(false);
+                    },
+                },
+                {
+                    'name'  : 'Cancel',
+                    'callback' : function() {
+                        deferred.reject();
+                        service.showPrompt(false);
+                        $location.url(homeUrl);
+                    },
+                }],
+                'responses' : responses
+            });
+        }
+
+        return deferred.promise;
+
+    };
+
+    /**
+     * Method to stop propagation of events.
+     *
+     * @param event
+     *     The event to stop.
+     */
+    var preventEvent = function(event) {
+        if (service.getPrompt())
+            event.stopPropagation();
+    };
+
+    /**
+     * When the prompt object is displayed, prevent keypress and
+     * mouse clicks from propagating through to the client so that
+     * the prompt dialog can be completed.
+     */
+    $window.addEventListener('input', preventEvent, true);
+    $window.addEventListener('keydown', preventEvent, true);
+    $window.addEventListener('keypress', preventEvent, true);
+    $window.addEventListener('keyup', preventEvent, true);
+    $window.addEventListener('click', preventEvent, false);
+
```

Comment:
And then there's this, which I think I need to replace with the keyboard disable stuff Mike sent me...